File: src/main/java/com/example/demoimdb/model/Actor.java
package com.example.demoimdb.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "actor")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Actor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "name")
    private String name;
    @Column(name = "description", length = 1000)
    private String description;
    @Column(name = "image")
    private String image;
    @Column(name = "dob")
    private String dob;
    @JsonIgnore
    @OneToMany(mappedBy = "actor")
    private List<MovieActor> listMovieActors;
}

File: src/entities/Actor.ts
import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from 'typeorm';
import { MovieActor } from './MovieActor';

@Entity({ name: 'actors' })
export class Actor {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @Column({ nullable: true })
  image: string;

  @Column({ nullable: true, length: 1000 })
  description: string;

  @Column({ nullable: true })
  dob: string;

  @OneToMany(() => MovieActor, (movieActor) => movieActor.actor)
  movieActors: MovieActor[];
}
